src/components/Activities/ListView.tsx

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+import { Activities } from "./Definition";
+
+interface PropsType<T> {
+  items: T[];
+  renderer: (item: T) => React.ReactNode;
+}
+
+export default function ListView<T extends Activities>(props: PropsType<T>) {
```

Comment:
Ok I figured out how to get the generic working fully so we could remove all idea of the Activity from the `ListView`:

Step 2: we need to remove the dependence on `Activities` inside this component, and define another way to make sure we have an `id` on the `item` object:

(no change to the interface, but deleting the first line and modifying the `extends` in the last line)

```suggestion
interface PropsType<T> {
  items: T[];
  renderer: (item: T) => React.ReactNode;
}

export default function ListView<T extends {id: number}>(props: PropsType<T>) {
```

Reply:
Done in 6f5ba69